plugin/core/transports.py

```diff
@@ -142,6 +142,8 @@ def _read_loop(self) -> None:
                     continue
 
                 def invoke(p: T) -> None:
+                    if self._closed:
+                        return
```

Comment:
Is this really correct? What if there's a final error message on stderr just before the transport closes? I'm not convinced this doesn't cause more trouble.

Reply:
This is the stdout, not stderr, loop. That said, stderr loop also already has the same check.